src/Microsoft.Health.Fhir.Shared.Api/Controllers/FhirController.cs

```diff
@@ -512,5 +513,30 @@ public async Task<IActionResult> BatchAndTransactions([FromBody] Resource bundle
 
             return FhirResult.Create(bundleResponse);
         }
+
+        /// <summary>
+        /// Returns resources defined in $everything operation
+        /// </summary>
+        /// <param name="idParameter">The resource ID</param>
+        /// <param name="start">The start date relates to care dates</param>
+        /// <param name="end">The end date relates to care dates</param>
+        /// <param name="since">Resources updated after this period will be included in the response</param>
+        /// <param name="type">Comma-delimited FHIR resource types to include in the return resources</param>
+        /// <param name="ct">The continuation token</param>
+        [HttpGet]
+        [Route(KnownRoutes.PatientEverythingById, Name = RouteNames.PatientEverythingById)]
+        [AuditEventType(AuditEventSubType.Everything)]
+        public async Task<IActionResult> PatientEverythingById(
```

Comment:
>PatientEverythingById [](http://example.com/codeflow?start=41&length=21)

[https://www.hl7.org/fhir/operations.html#executing](https://www.hl7.org/fhir/operations.html#executing)

[https://www.hl7.org/fhir/operations.html#request](https://www.hl7.org/fhir/operations.html#request)


I was always under impression what default behaviour for any operation is to have POST request with `Parameters` object inside body, and in some specific cases it's ok to have alternative of running GET request 

>If all the parameters for the operation are primitive types and the operation has affectsState = false, the operation may be invoked by performing an HTTP GET operation where all of the values of the parameters are appended to the URL in the search portion of the URL.


So I would add HttpPost action to controller as well.

Reply:
Thanks for sharing this. I've read through the specs. My understanding is that this operation has some characteristics that make it more suitable for GET.

(1) The input parameters can be empty, which is different from other operations like $convert-data or $member-match. The [spec](https://www.hl7.org/fhir/operations.html#executing) mentioned that:

> Executing operations without any parameters is a special case. For an operation that doesn't cause any state change, the operation is invoked in a straight forward fashion:
GET [base]/Composition/example/$document
For operations that call state changes, they must be invoked by a POST. There is no parameters resource in this case because a parameters resource cannot be empty. So the operation is invoked with a POST with an empty body:
POST [base]/Claim/example/$submit
Content-Length: 0

I think Patient $everything belongs to the first case that doesn't cause any state change. Besides, in current FHIR Server, posting empty body will result in bad request.

(2) This operation supports paging. A continuation token would be returned and `link.next` provides as a GET call for following pages.

(3) There are several things we need to consider before supporting POST, e.g., handling conflicts between parameters specified in URL and body.

So I would suggest enabling only GET first.